src/roman_datamodels/datamodels/_datamodels.py

```diff
@@ -130,6 +130,58 @@ class ImageModel(_RomanDataModel):
 class ScienceRawModel(_RomanDataModel):
     _node_type = stnode.WfiScienceRaw
 
+    @classmethod
+    def from_tvac_raw(cls, model):
+        """Convert TVAC/FPS into ScienceRawModel
+
+        Parameters
+        ----------
+        model : ScienceRawModel, TvacModel, FpsModel
+            Model to convert from.
+
+        Returns
+        -------
+        science_raw_model : ScienceRawModel
+            The ScienceRawModel built from the input model.
+            If the input was a ScienceRawModel, that model is simply returned.
+        """
+        ALLOWED_MODELS = (FpsModel, ScienceRawModel, TvacModel)
+
+        if isinstance(model, cls):
+            return model
+        if not isinstance(model, ALLOWED_MODELS):
+            raise ValueError(f"Input must be one of {ALLOWED_MODELS}")
+
+        # Create base ramp node with dummy values (for validation)
+        from roman_datamodels.maker_utils import mk_level1_science_raw
+
+        raw = mk_level1_science_raw(shape=model.shape)
+
+        # Define how to recursively copy all attributes.
+        def node_update(raw, other):
+            """Implement update to directly access each value"""
+            for key in other.keys():
```

Comment:
This is rather similar to:
https://github.com/spacetelescope/roman_datamodels/blob/main/src/roman_datamodels/datamodels/_datamodels.py#L177

and I seem to recall other spots in the code doing similar recursive updating. Maybe it is best to make a utility function that these methods can call?

Reply:
And methinks have accomplished it.